src/Middleware/AuthenticationMiddleware.php

```diff
@@ -85,6 +85,13 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
 
         try {
             $result = $service->authenticate($request);
+            $authenticator = $service->getAuthenticationProvider();
+
+            if ($authenticator !== null && !$authenticator instanceof StatelessInterface) {
+                assert($result->getData() !== null);
+                $service->persistIdentity($request, new Response(), $result->getData());
```

Comment:
You are ignored the return value of `persistIdentity()`.  When `CookieAuthenticator` is used the return value will contain the updated response instance with the header set for the cookie. So will your patch the cookie will never be set on the client.

Reply:
Nor you can return the response instance at this point as returning a response from the middleware will short circuit the response handling.

Reply:
OK, how can we fix this up to adhere to that?

Reply:
There's nothing that can be done here. PSR-15's middleware implementation doesn't support passing an updated response instance down the queue, it can only pass down the request.